File: Android/RTMSyncManager/src/main/java/io/agora/rtmsyncmanager/RoomExpirationPolicy.kt
package io.agora.rtmsyncmanager

/**
 * Class representing the room expiration policy.
 */
class RoomExpirationPolicy {
    /**
     * The expiration time of the room in milliseconds.
     * A value of 0 indicates that the room does not expire.
     */
    var expirationTime: Long = 0

    /**
     * A flag indicating whether the room's existence is associated with the owner's online status.
     * If true, the room is destroyed when the owner goes offline.
     * If false, the room persists even when the owner goes offline.
     */
    var isAssociatedWithOwnerOffline: Boolean = true

    /**
     * The maximum time in milliseconds that the owner can be offline before the room is destroyed.
     */
    var ownerReconnectMaxTime: Long = 5 * 60 * 1000
}

File: iOS/AgoraEntScenarios/Common/API/RTMSyncManager/RTMSyncManager/Classes/Syncmanager/RoomExpirationPolicy.swift
//
//  RoomExpirationPolicy.swift
//  RTMSyncManager
//
//  Created by wushengtao on 2024/5/6.
//

import Foundation

// 房间过期策略模型
public class RoomExpirationPolicy: NSObject {
    // 房间过期时间，单位ms，0表示房间不过期
    public var expirationTime: UInt64 = 0
    
    /// 房主离线最长时间，单位ms，超过该时间认为房间被销毁，0表示不处理
    public var ownerReconnectMaxTime: UInt64 = 5 * 60 * 1000
    
    // 是否和房主离线关联，true: 房主不在线则销毁房间，false: 房主不在线不销毁房间
    public var isAssociatedWithOwnerOffline: Bool = true
    
    public static func defaultPolicy() -> RoomExpirationPolicy {
        let policy = RoomExpirationPolicy()
        policy.expirationTime = 20 * 60 * 1000
        
        return policy
    }
}
